setup/terraform/resources/cdsw_setup.py

```diff
@@ -22,172 +22,255 @@
-    models = [m for m in r.json() if m['name'] == model_name]
-    if models and models[0]['latestModelDeployment']['status'] == 'deployed':
-        print('Model is already deployed!! Skipping.')
-        exit(0)
-
-    print('# Add engine')
-    r = s.post(CDSW_API + '/site/engine-profiles', json={'cpu': 1, 'memory': 4})
-    engine_id = r.json()['id']
-    print('Engine ID: %s'% (engine_id,))
-    
-    print('# Add environment variable')
-    r = s.patch(CDSW_API + '/site/config', json={'environment': '{"HADOOP_CONF_DIR":"/etc/hadoop/conf/"}'})
-    
-    print('# Add project')
-    project_name = 'Edge2AI Workshop'
-    project_id = None
-    r = s.get(CDSW_API + '/users/admin/projects')
-    for project in r.json():
-        if project['name'] == project_name:
-            project_id = project['id']
+
+s = requests.Session()
+
+print('# Create user')
+while True:
+    try:
+        r = s.post(CDSW_API + '/users', json={
+                'email': EMAIL,
+                'name': FULL_NAME,
+                'username': USERNAME,
+                'password': PASSWORD,
+                'type': 'user'
+            }, timeout=10)
+        if r.status_code == 201:
+            break
+        elif r.status_code == 422:
+            print('WARNING: User admin already exists')
             break
-    if not project_id:
-        r = s.post(CDSW_API + '/users/admin/projects', json={'template': 'git',
-                                                             'project_visibility': 'private',
-                                                             'name': project_name,
-                                                             'gitUrl': 'https://github.com/cloudera-labs/edge2ai-workshop'})
-        project_id = r.json()['id']
-    print('Project ID: %s'% (project_id,))
-    
-    print('# Upload setup script')
-    setup_script = """!pip3 install --upgrade pip scikit-learn
+    except requests.exceptions.ConnectTimeout:
+        pass
+    except requests.exceptions.ConnectionError:
+        pass
+    if r:
+        print('Waiting for CDSW to be ready... (error code: %s)' % (r.status_code,))
+    else:
+        print('Waiting for CDSW to be ready... (connection timed out)')
+    time.sleep(10)
+
+print('# Authenticate')
+r = s.post(CDSW_API + '/authenticate', json={'login': USERNAME, 'password': PASSWORD})
+s.headers.update({'Authorization': 'Bearer ' + r.json()['auth_token']})
+
+print('# Check if model is already running')
+model_name = 'IoT Prediction Model'
+r = s.post(CDSW_ALTUS_API + '/models/list-models', json={'projectOwnerName': 'admin', 'latestModelDeployment': True, 'latestModelBuild': True})
+models = [m for m in r.json() if m['name'] == model_name]
+if models and models[0]['latestModelDeployment']['status'] == 'deployed':
+    print('Model is already deployed!! Skipping.')
+    #exit(0)
```

Comment:
In case the model has already been deployed, we don't want to re-run the steps below either (Add engine, variables, etc). At this point we could jump directly to the beginning of the VizApps section.

It would be better to encapsulate the model deployment and the vizapps deployment in separate functions and call these functions from the main script section.

This `exit(0)` would turn into a `return`, so that the whole `deploy_model` function would terminate and the `deploy_vizapps` one would be executed.